webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -82,28 +85,103 @@ export class MonacoService {
+              //network load or switched to currently open file
+              console.log(`ZWED0120I network load or switched to currently open file`);              
+              const resJson = response;
+              this.setMonacoModel(fileNode, <{ contents: string, language: string }>resJson).subscribe({
+                next: () => {
+                  this.editorControl.fileOpened.next({ buffer: fileNode, file: fileNode.name });
+                  if (line) {
+                    this.editorControl.editor.getValue().revealPosition({ lineNumber: line, column: 0 });
+                    this.decorations.push(this.editorControl.editor.getValue().deltaDecorations([], [
+                      { range: new monaco.Range(line, 100, line, 100), options: { isWholeLine: true, inlineClassName: 'highlight-line' } },
+                    ])[0]);
+                    // this.editor.getValue().colorizeModelLine(newModel, fileNode.model.line);
+                  }
+                  if (reload) {
+                    this.editorControl.initializedFile.next(fileNode);
+                  }
+                },
+                error: (err) => {
+                  this.log.warn(err);
+                }
+              });
+            },
+            error: (err) => {
+              console.log(`ZWED0142I could not be accessed`);
+              this.log.warn(`${fileNode.name} could not be accessed, status: `, err.status);
+              
+              if (err.status === 403) {
+                this.snackBar.open(`${fileNode.name} could not be accessed due to permissions.`,
+                  'Close', { duration: MessageDuration.Medium, panelClass: 'center' });
+              } else if (err.status === 404) {
+                this.snackBar.open(`${fileNode.name} lock is not available.`,
+                  'Close', { duration: MessageDuration.Medium, panelClass: 'center' });
+              } else {
+                this.snackBar.open(`${fileNode.name} other error 152.`,
+                  'Close', { duration: MessageDuration.Medium, panelClass: 'center' });
+              }
+            }
+          });
+          /* end new code for ENQ */
+          console.log(`openFile ended 157`); 
+          return; /* we have dealt with reloading datasets, so we can quit */
+
+        } else /* it's not a dataset */ {
           requestUrl = ZoweZLUX.uriBroker.unixFileUri('contents',
                                                       filePath+'/'+fileNode.model.fileName,
                                                       { responseType: 'b64' });
           _observable = this.http.get(requestUrl).map((res: any) => this.dataAdapter.convertFileContent(res._body));
         }
 
-      } else {
+      } else { /* not reloading */
         _observable = new Observable((obs) => obs.next({ contents: fileNode.model.contents }));
       }
+      
+      /* this is always executed, reloading or not,
+          but not when reloading a dataset. */
```

Comment:
I see, ok. This makes sense via the return above

Reply:
There's probably a cleaner way to do this.  Deferred.